pkg/utils/utils.go

```diff
@@ -476,6 +478,11 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 				}
 			}
 
+			// Do not configure VF which is not part of any VF Group.
+			if !found {
+				continue
+			}
```

Comment:
This is essentially filtering the list of addresses to process. Now, the body of the loop does much too much imo. Let's take the opportunity to refactor this whole function to have a more proper way to filter the list

Reply:
done, please have a look again